disnake/flags.py

```diff
@@ -161,6 +163,34 @@ def _set_flag(self, o: int, toggle: bool) -> None:
             raise TypeError(f"Value to set for {self.__class__.__name__} must be a bool.")
 
 
+class ListBaseFlags(BaseFlags):
+    # This is 1-indexed, i.e. value `1` corresponds to bit 0, `2` corresponds to bit 1, ...
+
+    __slots__ = ()
+
+    @classmethod
+    def _from_values(cls, values: Sequence[int]):
+        self = cls.__new__(cls)
+        value = 0
+        for n in values:
+            # protect against DoS with large shift values
+            # n.b. performance overhead of this is negligible
+            if not (0 < n < 64):
+                raise ValueError("Flag values must be within (0, 64)")
+            value += 1 << (n - 1)
+        self.value = value
+        return self
+
+    @property
+    def values(self) -> List[int]:
+        # this may look weird but interestingly it's by far the
+        # fastest out of all benchmarked snippets
+        return [i for i, c in enumerate(bin(self.value)[:1:-1], 1) if c == "1"]
```

Comment:
this should document *what* this snippet does and additionally IMO, the other code that was slower

Reply:
Good point, added some more info in c4a4f517ba9b31b44ee4d852710a6af346eb8c61, does that clarify it?

Reply:
Yes